subsys/net/lib/dns/Kconfig

```diff
@@ -3,6 +3,7 @@
 
 config DNS_RESOLVER
 	bool "DNS resolver"
+	depends on NET_NATIVE
```

Comment:
DNS resolver should work ok if `CONFIG_NET_OFFLOAD` is set because the offloading is happening under `net_context` API which the resolver is using. Resolver is not supported if `CONFIG_NET_SOCKETS_OFFLOAD` is set.
So I think we should have here
```
depends on NET_NATIVE || NET_OFFLOAD
```

Reply:
DNS resolver setup fails when binding local address, it fails as context is not ready and it is not ready as there is no interface. And there is no interface because for non native stack it is always NULL.

https://github.com/zephyrproject-rtos/zephyr/blob/main/include/net/net_if.h#L1782

So far it does not work and it would be great to add dependency to avoid people wasting time looking why.

Reply:
Hmm, indeed you have a point. I just wonder how people are using the DNS resolver when (non-socket) offloading is enabled even if it in theory could be done.